modules/web/js/ballerina/diagram/views/default/components/annotation-definition.jsx

```diff
@@ -17,26 +17,211 @@
+import Alerts from 'alerts';
+import _ from 'lodash';
+import { getCanvasOverlay } from 'ballerina/configs/app-context';
+import AnnotationDefinitionAST from 'ballerina/ast/annotation-definition';
+import ASTFactory from 'ballerina/ast/ast-factory.js';
 import PanelDecorator from './panel-decorator';
-import { getComponentForNodeArray } from './../../../diagram-util';
+import { getComponentForNodeArray, getDesigner } from './../../../diagram-util';
 import AnnotationAttributeDecorator from './annotation-attribute-decorator';
 import AnnotationDefinitionAttachment from './annotation-definition-attachment';
 import TagController from './utils/tag-component';
-import Alerts from 'alerts';
-import _ from 'lodash';
-import PropTypes from 'prop-types';
+import ActionMenu from './action-menu';
 
+/**
+ * React view for annotation defintion.
+ * @class AnnotationDefinition
+ * @extends {React.Component}
+ */
 class AnnotationDefinition extends React.Component {
 
+    /**
+     * Creates an instance of AnnotationDefinition.
+     * @memberof AnnotationDefinition
+     */
     constructor() {
         super();
         this.onAttachmentDelete = this.onAttachmentDelete.bind(this);
         this.addAttachmentPoint = this.addAttachmentPoint.bind(this);
     }
 
     /**
-     * Render view for annotation definition
+     * @override
+     */
+    componentDidMount() {
+        if (this.props.model.getViewState().collapsed === false) {
+            this.createActionMenu();
+        }
+    }
+
+    /**
+     * @override
+     */
+    componentDidUpdate() {
+        if (this.actionMenuWrapper) {
+            ReactDOM.unmountComponentAtNode(this.actionMenuWrapper);
+            const canvasOverlay = getCanvasOverlay();
+            if (canvasOverlay.contains(this.actionMenuWrapper)) {
+                canvasOverlay.removeChild(this.actionMenuWrapper);
+            }
+            if (this.props.model.getViewState().collapsed === false) {
+                this.createActionMenu();
+            }
+        }
+    }
+
+    /**
+     * Delete attachment tag event handler.
+     * @param {string} attachment - attachment to be deleted
+     * */
+    onAttachmentDelete(attachment) {
+        const model = this.props.model;
+        delete model.getViewState().attachments[attachment];
+        model.removeAnnotationAttachmentPoints(attachment);
+    }
+
+    /**
+     * Get types of ballerina to which can be applied when declaring variables.
+     * @return {[string]} Attachment types
+     * */
+    getTypeDropdownValues() {
+        const { environment } = this.context;
+        return environment.getAnnotationAttachmentTypes();
+    }
+
+    /**
+     * Gets the suggestions for attachment points for the definition.
+     * @returns {string[]} The suggestions.
+     * @memberof AnnotationDefinition
+     */
+    getAnnotationAttachmentPointsForSuggestions() {
+        const { environment } = this.context;
+        const suggestions = [];
```

Comment:
`const attachmentTypes = environment.getAnnotationAttachmentTypes();` will look much nicer since it's used twice.